From 482bce6ca3a6edbe19e127a216010df270832382 Mon Sep 17 00:00:00 2001
From: Corey Minyard <cminyard@mvista.com>
Date: Fri, 10 Feb 2017 22:39:16 -0600
Subject: [PATCH] MIPS:kdump: Add some more symbols

Support for 4-level page tables and 48-bit address spaces is coming.
Add symbols so these can be detected and handled properly.

Also, we need to include asm/sections.h to get _text and _end.

Signed-off-by: Corey Minyard <cminyard@mvista.com>
---
 arch/mips/kernel/machine_kexec.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/arch/mips/kernel/machine_kexec.c b/arch/mips/kernel/machine_kexec.c
index 299ed22..1193012 100644
--- a/arch/mips/kernel/machine_kexec.c
+++ b/arch/mips/kernel/machine_kexec.c
@@ -13,6 +13,7 @@
 #include <asm/cacheflush.h>
 #include <asm/page.h>
 #include <asm/pgtable.h>
+#include <asm/sections.h> /* for _text, _end */
 
 extern const unsigned char relocate_new_kernel[];
 extern const size_t relocate_new_kernel_size;
@@ -118,7 +119,10 @@ void arch_crash_save_vmcoreinfo(void)
 {
 	VMCOREINFO_NUMBER(PAGE_SHIFT);
 	VMCOREINFO_NUMBER(PGD_ORDER);
-#if defined(CONFIG_32BIT) || !defined(CONFIG_PAGE_SIZE_64KB)
+#if !defined(__PAGETABLE_PUD_FOLDED)
+	VMCOREINFO_NUMBER(PUD_ORDER);
+#endif
+#if !defined(__PAGETABLE_PMD_FOLDED)
 	VMCOREINFO_NUMBER(PMD_ORDER);
 #endif
 	VMCOREINFO_NUMBER(PTE_ORDER);
@@ -127,10 +131,12 @@ void arch_crash_save_vmcoreinfo(void)
 	VMCOREINFO_NUMBER(_PAGE_HUGE);
 #endif
 #ifdef CONFIG_64BIT
+	VMCOREINFO_NUMBER(TASK_SIZE64);
 #ifdef CONFIG_MAPPED_KERNEL
 	VMCOREINFO_ADDRESS(_text);
 	VMCOREINFO_ADDRESS(_end);
 	VMCOREINFO_ADDRESS(phys_to_kernel_offset);
+	VMCOREINFO_ADDRESS(kernel_image_end);
 #endif
 	VMCOREINFO_ADDRESS(CKSEG0);
 	VMCOREINFO_ADDRESS(CKSSEG);
@@ -140,5 +146,6 @@ void arch_crash_save_vmcoreinfo(void)
 
 	VMCOREINFO_ADDRESS(PAGE_OFFSET);
 	VMCOREINFO_ADDRESS(IO_BASE);
+	VMCOREINFO_ADDRESS(MAP_BASE);
 	VMCOREINFO_NUMBER(_PFN_SHIFT);
 }
-- 
2.7.4

